src/linear_algebra/std_basis.lean

```diff
@@ -158,10 +158,9 @@ variables {R : Type*}
 
 section module
 variables {η : Type*} {ιs : η → Type*} {Ms : η → Type*}
-variables [ring R] [∀i, add_comm_group (Ms i)] [∀i, module R (Ms i)]
 
-lemma linear_independent_std_basis [decidable_eq η]
-  (v : Πj, ιs j → (Ms j)) (hs : ∀i, linear_independent R (v i)) :
+lemma linear_independent_std_basis [ring R] [∀i, add_comm_group (Ms i)] [∀i, module R (Ms i)]
```

Comment:
This file is essentially the same as before, i just generalized it to `semiring`, except `linear_independent_std_basis` that does not work. I didn't touch any proof.